### PR TITLE
feat(s3): allow consumers to specify bucket website URL format

### PR DESCRIPTION
**Pull request by AlexChesters**
----

Opened after a discussion with @rix0rrr after #1544 was merged.

### Pull Request Checklist

* [ not needed ] Testing
  - Unit test added
  - __CLI change?:__ manually run [integration tests](https://github.com/awslabs/aws-cdk/blob/master/packages/aws-cdk/integ-tests/test.sh) and paste output as a PR comment
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [X] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [X] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ not needed ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

**Comment by AlexChesters** [comment]
@rix0rrr if I understand what I've done correctly, I _think_ this PR may only change `ImportedBucket` rather than the "normal" `Bucket` construct. Am I correct? If so, I couldn't quite figure out how to implement this for the "normal" `Bucket` so hopefully you or someone else can chime in with some advice 😄

**Comment by AlexChesters** [comment]
I've marked the `Testing` section on the PR template as "not needed" as this change is already covered by existing test cases. Happy to be convinced otherwise though 😄 

Indeed depending on any replies to [my comment above](https://github.com/awslabs/aws-cdk/pull/1550#issuecomment-454395535) there may be some additional changes to this PR that do require new tests added.

File: packages/@aws-cdk/aws-s3/lib/bucket.ts [hunk 1]
```diff
@@ -1000,6 +1012,8 @@ class ImportedBucket extends BucketBase {
   }
 
   private generateBucketWebsiteUrl() {
-    return `${this.bucketName}.s3-website-${new cdk.Aws().region}.amazonaws.com`;
+    return this.bucketWebsiteNewUrlFormat
+      ? `${this.bucketName}.s3-website-${cdk.Stack.find(this).region}.${cdk.Stack.find(this).urlSuffix}`
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-s3/lib/bucket.ts]: It is just the other way around; new format is with `.`, old format is with `-`.

I support the default being `false`.

**Comment by rix0rrr** [comment]
And thank you for submitting this PR!

**Comment by AlexChesters** [comment]
@rix0rrr not a problem 😄 

I've pushed two changes, one to actually make sure the new format is correct (i.e. with a `.`, not a `-`) and one to default the prop to `false` as per your suggestion - how do they look?

I agree with you that it feels like there should be a test here, have you got any ideas? Currently it's a `private` function so I'm not sure how this test would look. I don't see how it would be possible by asserting against the generated stack as it just eventually outputs a `Fn::GetAtt` snippet rather than the actual value.

**Comment by rix0rrr** [comment]
You're right. It's only for imported buckets, so is not bound to come up often. Thanks!

**Pull request merged**